internal/rest/resources/control.go

```diff
@@ -238,11 +242,11 @@ func joinWithToken(state state.State, r *http.Request, req *internalTypes.Contro
 		}
 
 		logger.Error("Unable to complete cluster join request", logger.Ctx{"address": addr.String(), "error": err})
-		lastErr = err
+		allErrors = append(allErrors, err)
 	}
 
 	if joinInfo == nil {
-		return nil, fmt.Errorf("%d join attempts were unsuccessful. Last error: %w", len(token.JoinAddresses), lastErr)
```

Comment:
I would vote against `errors.Join` as it simply concatenates the errors using a newline which looks odd in my opinion as we use the pattern `Error: Error more specific: Error more specific`.

Overall I think the idea is to only show the actual error from the join process itself. Let's consider `shared.GetRemoteCertificate` returns an error three times for all the join tokens addresses. It prints three warnings but `joinInfo` is still set to `nil`.
So probably its best to populate `lastErr` every time we log a warning.

Reply:
Well, ``errors.Join``, doesn't simply concatenate the error messages, ``errors.Is`` can still be used on the resulting error to match any of its components. But ultimately, it won't be relevant after marshaling the error.

Reply:
I see what you are saying but we usually don't have multiline errors and tend to have everything in a single line.